engine/sawtooth_poet_engine/engine.py

```diff
@@ -194,7 +196,7 @@ def start(self, updates, service, startup_state):
                 LOGGER.exception("Unhandled exception in message loop")
 
     def _try_to_publish(self):
-        if self._published:
+        if self._published or self._validating_blocks:
```

Comment:
I don't see a direct benefit here, don't know if results prove other way. It could be a block from another validator and at any height. As per the expected behavior, now the block's validity too isn't verified yet by the validator. Also at this stage the PoET is unsure of whether to go with the received block or there's possibility for it to end up in an unused fork.

Waiting at this point in time could be expensive.

This optimization applies if PoET does intelligent fork resolution, i.e. it at least should know whether this particular block's has chances of winning the election. Part of fork resolution algorithm can be used here to determine the winning possibility.

Reply:
Deferring local publishing until the backlog is processed was one of the main drivers for this change.

Reply:
It'll be interesting to see Z test behavior with this patch.